src/bsort.py

```diff
@@ -47,25 +55,46 @@ def count_sort(x: list[int]) -> list[int]:
     >>> count_sort([1, 2, 1, 2, 4])
     [1, 1, 2, 2, 4]
     """
+    # Count-sort where the output list have the final length to begin
+    # with.
     counts = count_keys(x)
-    out = [0] * len(x)
-    # FIXME: do the actual sorting
+    out = [0]*len(x)
+    j = 0
+    for key in range(max(x)+1): # Iterate over all keys. 
+        for i in range(j, counts[key]+j):
+            out[i]=key
+        j += counts[key]
+
+    # Simple count-sort where output list is growing. 
+    #counts = count_keys(x)
+    #out = [] # otherwise out = [0] * len(x)
+    #for key, count in enumerate(counts):
+    #    out.extend([key]*count)
     return out
 
+print('Count sort:')
+print(count_sort([1, 2, 1, 2, 4]))
+
 
 def cumsum(x: list[int]) -> list[int]:
     """
     Calculate the cumulative sum of x.
 
     >>> cumsum([1, 2, 3])
-    [0, 1, 3]
+    [0, 1, 3] # Hvis cumulative sum, burde det ikke være [1, 3, 6]?
     >>> cumsum([0, 2, 2, 0, 1])
-    [0, 0, 2, 4, 4]
+    [0, 0, 2, 4, 4] # Bør dette ikke være [0, 2, 4, 4, 5]? Så ville det
```

Comment:
Vores input betyder at vi har 0 0'ere, 2 1'er, 2 2'er, 0 3'er og 1 4'er.

```
  [1, 1, 2, 2, 4]
```

De nøgler vi ikke har kan du tænke på som liggende som tomme blokke (der starter og slutter på samme indeks).

```
  [<tom 0 blok> <1 blok: 1, 1>, <2 blok: 2, 2>, <tom 3 blok>, <4 blok: 4>]
```

og vi skal returnere hvor hver enkel nøgles blok starter:

```
     [<tom 0 blok> <1 blok: 1, 1>, <2 blok: 2, 2>, <tom 3 blok>, <4 blok: 4>]
0 ----^ (index 0)
1 ----------------^ (også indeks 0 da foregående blok er tom)
2 ---------------------------------^ (indeks 2 da der er to 1'ere før blokken)
3 -------------------------------------------------^ (indeks 4 da vi har [1,1,2,2] før dette punkt)
4 ---------------------------------------------------------------^ (indeks 4 da 3-blokken er tom)
```

Det er indeks for nøglerne 0, 1, 2, 3, og 4 vi skal returnere, altså `[0, 0, 2, 4, 4]`